### PR TITLE
chore: bump Example apps to RN 0.73.2

### PR DESCRIPTION
**Pull request by WoLewicki**
## Summary

PR bumping `Example` and `FabricExample` apps to RN v0.73.2 :rocket:

## Test plan

Build apps and click through them.

### CODE REVIEW

File: .github/workflows/ios-build.yml [hunk 1]
```diff
@@ -85,4 +85,4 @@ jobs:
 
       - name: Build app
         working-directory: ${{ matrix.working-directory }}
-        run: npx react-native run-ios
+        run: npx react-native@latest run-ios --simulator="iPhone 14 Pro" --mode Debug --verbose --terminal /bin/zsh
```
Comment (tomekzaw) [inline .github/workflows/ios-build.yml:88]: Why is this change necessary?

File: .github/workflows/validate-plugin.yml [hunk 2]
```diff
@@ -23,10 +23,10 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Use Node.js 16
+      - name: Use Node.js 18
         uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 18
```
Comment (tomekzaw) [inline .github/workflows/validate-plugin.yml:29]: Should we add `.nvmrc` file with predefined version of Node.js to be used in the project?

File: Example/android/app/build.gradle [hunk 3]
```diff
@@ -52,16 +51,6 @@ react {
     // hermesFlags = ["-O", "-output-source-map"]
 }
 
-project.ext["isReanimatedExampleApp"] = true
```
Comment (tomekzaw) [inline Example/android/app/build.gradle:55]: Is it safe to remove this line?

File: .github/workflows/build-npm-package-action.yml [hunk 4]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Setup Node
```
Comment (tomekzaw) [inline .github/workflows/build-npm-package-action.yml:32]:
```suggestion
      - name: Use Node.js 18
```

File: Example/android/app/build.gradle [hunk 5]
```diff
@@ -80,38 +69,19 @@ def enableProguardInReleaseBuilds = false
...
-
 android {
     ndkVersion rootProject.ext.ndkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
+    compileSdk rootProject.ext.compileSdkVersion
 
-    compileSdkVersion rootProject.ext.compileSdkVersion
-
-    namespace "com.swmansion.reanimated.example"
+    namespace "com.example"
```
Comment (tomekzaw) [inline Example/android/app/build.gradle:77]: Do we really want to change the namespace here? I'm okay with that change (it's better to have our app aligned our app with the official template).

File: Example/android/app/build.gradle [hunk 6]
```diff
@@ -80,38 +69,19 @@ def enableProguardInReleaseBuilds = false
...
     defaultConfig {
-        applicationId "com.swmansion.reanimated.example"
+        applicationId "com.example"
```
Comment (tomekzaw) [inline Example/android/app/build.gradle:79]: Do we really want to change the application id here? I'm okay with that change (it's better to have our app aligned our app with the official template).

File: Example/package.json [hunk 7]
```diff
@@ -11,44 +11,43 @@
...
-    "react-native-pager-view": "^5.4.25",
+    "react-native-gesture-handler": "^2.14.1",
+    "react-native-pager-view": "^6.2.3",
     "react-native-reanimated": "link:../",
-    "react-native-safe-area-context": "^4.7.2",
-    "react-native-screens": "^3.27.0",
-    "react-native-svg": "^13.14.0"
+    "react-native-safe-area-context": "^4.8.2",
+    "react-native-screens": "^3.29.0",
+    "react-native-svg": "^14.1.0"
```
Comment (tomekzaw) [inline Example/package.json:29]: Thanks for bumping the libraries as well 🙏

File: Example/patches/metro-inspector-proxy+0.76.8.patch [hunk 8]
```diff
@@ -1,177 +0,0 @@
-diff --git a/node_modules/metro-inspector-proxy/src/Device.js b/node_modules/metro-inspector-proxy/src/Device.js
```
Comment (tomekzaw) [inline Example/patches/metro-inspector-proxy+0.76.8.patch:1]: Does debugging with Chrome DevTools still work fine if this patch is removed?

File: Example/patches/react-native-box2d+0.2.5.patch [hunk 9]
```diff
@@ -72,7 +72,17 @@ index 41726f4..0580d5d 100644
    defaultConfig {
      minSdkVersion 21
      targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
-@@ -342,8 +344,8 @@ def nativeBuildDependsOn(dependsOnTask, variant) {
+@@ -176,7 +178,8 @@ def thirdPartyVersionsFile = new File("${androidSourcesDir.toString()}/ReactAndr
+ def thirdPartyVersions = new Properties()
+ thirdPartyVersions.load(new FileInputStream(thirdPartyVersionsFile))
+ 
+-def BOOST_VERSION = thirdPartyVersions["BOOST_VERSION"]
++
```
Comment (tomekzaw) [inline Example/patches/react-native-box2d+0.2.5.patch:80]: Let's remove this unnecessary newline (probably you'll need to re-generate the patch).

File: FabricExample/android/app/build.gradle [hunk 10]
```diff
@@ -52,16 +51,6 @@ react {
     // hermesFlags = ["-O", "-output-source-map"]
 }
 
-project.ext["isReanimatedExampleApp"] = true
```
Comment (tomekzaw) [inline FabricExample/android/app/build.gradle:55]: Is it safe to remove this line?

File: FabricExample/ios/FabricExample/Info.plist [hunk 11]
```diff
@@ -38,7 +42,19 @@
 		</dict>
 	</dict>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string/>
+	<string></string>
```
Comment (tomekzaw) [inline FabricExample/ios/FabricExample/Info.plist:45]: Why not `<string/>`?

File: FabricExample/patches/metro-inspector-proxy+0.76.8.patch [hunk 12]
```diff
@@ -1,177 +0,0 @@
-diff --git a/node_modules/metro-inspector-proxy/src/Device.js b/node_modules/metro-inspector-proxy/src/Device.js
```
Comment (tomekzaw) [inline FabricExample/patches/metro-inspector-proxy+0.76.8.patch:1]: Same question as for Example app

File: FabricExample/patches/react-native-box2d+0.2.5.patch [hunk 13]
```diff
@@ -72,7 +72,17 @@ index 41726f4..0580d5d 100644
    defaultConfig {
      minSdkVersion 21
      targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
-@@ -342,8 +344,8 @@ def nativeBuildDependsOn(dependsOnTask, variant) {
+@@ -176,7 +178,8 @@ def thirdPartyVersionsFile = new File("${androidSourcesDir.toString()}/ReactAndr
+ def thirdPartyVersions = new Properties()
+ thirdPartyVersions.load(new FileInputStream(thirdPartyVersionsFile))
+ 
+-def BOOST_VERSION = thirdPartyVersions["BOOST_VERSION"]
++
```
Comment (tomekzaw) [inline FabricExample/patches/react-native-box2d+0.2.5.patch:80]: Same suggestion as for Example app

**Review by tomekzaw — CHANGES_REQUESTED** [review]

File: .github/workflows/ios-build.yml (hunk 1 above)
Comment (WoLewicki) [reply .github/workflows/ios-build.yml:88]: Without `--terminal /bin/zsh` specified, the job fails at the beginning with message that no terminal has been specified 😮  `@latest` is recommended by Meta team iirc, and `--mode` and `--verbose` are not necessary but can be helpful with debugging most probably.

File: .github/workflows/validate-plugin.yml (hunk 2 above)
Comment (WoLewicki) [reply .github/workflows/validate-plugin.yml:29]: No

File: Example/android/app/build.gradle (hunk 3 above)
Comment (WoLewicki) [reply Example/android/app/build.gradle:55]: Hmm I didn't spot that it has been removed, but the app seems to work correctly without it so 🤷‍♂️

File: Example/android/app/build.gradle (hunk 5 above)
Comment (WoLewicki) [reply Example/android/app/build.gradle:77]: Yeah, I didn't like all these shenanigans needed to change `Example` to `ReanimatedExample` on both `iOS` and `Android` so I'd remove it as a whole.

File: Example/android/app/build.gradle (hunk 6 above)
Comment (WoLewicki) [reply Example/android/app/build.gradle:79]: same

File: Example/package.json (hunk 7 above)
Comment (WoLewicki) [reply Example/package.json:29]: no problemo

File: Example/patches/metro-inspector-proxy+0.76.8.patch (hunk 8 above)
Comment (WoLewicki) [reply Example/patches/metro-inspector-proxy+0.76.8.patch:1]: There is no such package in `node_modules` right now and I didn't check. @piaskowyk declared that you will take care of it later, am I right?

File: Example/patches/react-native-box2d+0.2.5.patch (hunk 9 above)
Comment (WoLewicki) [reply Example/patches/react-native-box2d+0.2.5.patch:80]: Do we really need to do it?

File: FabricExample/ios/FabricExample/Info.plist (hunk 11 above)
Comment (WoLewicki) [reply FabricExample/ios/FabricExample/Info.plist:45]: It's generated by Xcode or app template so idk

File: FabricExample/android/app/build.gradle (hunk 10 above)
Comment (WoLewicki) [reply FabricExample/android/app/build.gradle:55]: Same as above.

File: Example/android/app/build.gradle (hunk 3 above)
Comment (tomekzaw) [reply Example/android/app/build.gradle:55]: The purpose of this flag is to resolve react-native directory to the one in example app's node_modules so this was added on purpose. We will need a stronger confirmation than "it works now".

File: Example/android/app/build.gradle (hunk 5 above)
Comment (tomekzaw) [reply Example/android/app/build.gradle:77]: Okay, this makes perfect sense. Thanks

File: Example/patches/react-native-box2d+0.2.5.patch (hunk 9 above)
Comment (tomekzaw) [reply Example/patches/react-native-box2d+0.2.5.patch:80]: not really but doesn't sound like much effort

File: Example/android/app/build.gradle (hunk 3 above)
Comment (WoLewicki) [reply Example/android/app/build.gradle:55]: Ok so I'll revert it

**Review by tomekzaw — APPROVED** [review]
@WoLewicki Thanks for applying code review suggestions!

Overall the PR looks good to me, although I haven't run it locally.
